x/ibc/testing/chain.go

```diff
@@ -791,14 +791,31 @@ func (chain *TestChain) SendPacket(
 	return nil
 }
 
-// ReceiveExecuted simulates receiving and writing an acknowledgement to the chain.
-func (chain *TestChain) ReceiveExecuted(
+// WriteReceipt simulates receiving and writing a receipt to the chain.
+func (chain *TestChain) WriteReceipt(
 	packet exported.PacketI,
 ) error {
 	channelCap := chain.GetChannelCapability(packet.GetDestPort(), packet.GetDestChannel())
 
 	// no need to send message, acting as a handler
-	err := chain.App.IBCKeeper.ChannelKeeper.ReceiveExecuted(chain.GetContext(), channelCap, packet, TestHash)
+	err := chain.App.IBCKeeper.ChannelKeeper.WriteReceipt(chain.GetContext(), channelCap, packet)
+	if err != nil {
+		return err
+	}
+
+	// commit changes
+	chain.App.Commit()
```

Comment:
Is doing this ok? @colin-axner

Reply:
I think this is ok as it represents the asynchrony of the acknowledgment?

Reply:
it's ok because no message is being sent to the chain (we are simulating one) and this function is called by the coordinator which immediately increments time afterwards. It isn't great design though (my bad) because it is counter-intuitive. It must be done though because otherwise the state change won't be updated on the client